src/components/date-picker/date-picker.ts

```diff
@@ -0,0 +1,792 @@
+              <igc-icon
+                name="clear"
+                collection="internal"
+                aria-hidden="true"
+              ></igc-icon>
+            </slot>
+          </span>
+        `;
+  }
+
+  private renderCalendarIcon() {
+    const defaultIcon = html`
+      <igc-icon
+        name="calendar_today"
+        collection="internal"
+        aria-hidden="true"
+      ></igc-icon>
+    `;
+
+    const state = this.open ? 'calendar-icon-open' : 'calendar-icon';
+
+    return html`
+      <span slot="prefix" part=${state} @click=${this.handleAnchorClick}>
+        <slot name=${state}>${defaultIcon}</slot>
+      </span>
+    `;
+  }
+
+  private renderCalendar(id: string) {
+    const hideHeader = this.isDropDown ? true : this.hideHeader;
+
+    return html`
+      <igc-calendar
+        aria-labelledby=${id}
+        role="dialog"
+        .inert=${!this.open || this.disabled}
+        ?show-week-numbers=${this.showWeekNumbers}
+        ?hide-outside-days=${this.hideOutsideDays}
+        ?hide-header=${hideHeader}
+        .activeDate=${this.activeDate ?? this.value ?? new Date()}
+        .value=${this.value ?? undefined}
+        .headerOrientation=${this.headerOrientation}
+        .orientation=${this.orientation}
+        .visibleMonths=${this.visibleMonths}
+        .locale=${this.locale}
+        .disabledDates=${this._dateConstraints!}
+        .specialDates=${this.specialDates}
+        .weekStart=${this.weekStart}
+        @igcChange=${this.handleCalendarChangeEvent}
+        exportparts="header, header-title, header-date, content: calendar-content, navigation, months-navigation,
+        years-navigation, years-range, navigation-buttons, navigation-button, days-view-container,
+        days-view, months-view, years-view, days-row, label: calendar-label, week-number, week-number-inner, date,
+        date-inner, first, last, inactive, hidden, weekend, range, special, disabled, single, preview,
+        month, month-inner, year, year-inner, selected, current"
+      >
+        ${!this.isDropDown
+          ? html`<slot name="title" slot="title">Select date</slot> `
+          : nothing}
+      </igc-calendar>
+    `;
+  }
+
+  protected renderActions() {
+    const slot = this.isDropDown || !this.actions.length ? undefined : 'footer';
+
+    // If in dialog mode use the dialog footer slot
+    return html`
+      <div
+        part="actions"
+        ?hidden=${!this.actions.length}
+        slot=${ifDefined(slot)}
+      >
+        <slot name="actions"></slot>
+      </div>
+    `;
+  }
+
+  protected renderPicker(id: string) {
+    return this.isDropDown
+      ? html`
+          <igc-popover ?open=${this.open} anchor=${id} flip shift>
+            <igc-focus-trap ?disabled=${!this.open || this.disabled}>
+              ${this.renderCalendar(id)}${this.renderActions()}
+            </igc-focus-trap>
+          </igc-popover>
+        `
+      : html`
+          <igc-dialog
+            aria-label="Select date"
```

Comment:
This should ideally be localized :)
Perhaps, we can re-use the same string from the resource strings at least, seems like it'll make _some_ sense.